src/Sarif.Multitool.Library/Rules/ADO1014.ProvideRequiredRunProperties.cs

```diff
@@ -34,16 +43,14 @@ protected override void Analyze(Run run, string runPointer)
                     // {0}: This 'run' object does not provide an 'automationDetails' property. This property is required by the {1} service.
                     LogResult(
                         runPointer,
-                        nameof(RuleResources.ADO1014_AdoProvideRequiredRunPropertties_Error_MissingAutomationDetails_Error_Text),
-                        this.ServiceName);
```

Comment:
>this.ServiceName [](http://example.com/codeflow?start=24&length=16)

When I debug I see you auto inject the ServiceName into position 1 when the ServiceName is not null, and passing it again results in extra ServiceName parameter at the end.
I am not sure how you want to fix this, do you want to keep the auto inject and remove the duplicate here?

though I am not sure if the auto inject is perfect, will it inject when it is not needed?

Reply:
I don't understand what you mean by "passing it again"

Reply:
Please see below step by step change of the var you will understand, you only need 2 in the end, but you got 3, the 2 and 3 are duplicated:

![image](https://github.com/microsoft/sarif-sdk/assets/81775155/43376374-9955-4d60-9157-279fd0c480de)

![image](https://github.com/microsoft/sarif-sdk/assets/81775155/06054085-0226-45e9-a2f9-df5fa41d0567)

![image](https://github.com/microsoft/sarif-sdk/assets/81775155/56ad8569-2ad2-4c0f-8de0-969ca8ee0007)

Reply:
Sigh...I don't remember the sequence of these changes. I'll try to refresh my memory...

Reply:
We can work this out in the next change